source/agora/node/Validator.d

```diff
@@ -685,7 +685,7 @@ public class Validator : FullNode, API
     {
         PreImageInfo preimage;
         if (this.enroll_man.getNextPreimage(preimage,
-            this.ledger.height()))
+            max(this.ledger.height(), this.ledger.expectedHeight(this.clock.utcTime()))))
```

Comment:
I think this is what is causing the recent `ci` failures. If the expected height is beyond the end of the enrollment cycle then `getNextPreimage` returns `ulong.max` as there is no enrollment at this height yet. Rather than revert this commit I will create a PR to fix it.